examples/finding_best_open_source_embedding_model/best_embedding_model_rag_app.ipynb

```diff
@@ -0,0 +1,559 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Finding the best open-source embedding model for your RAG application"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this tutorial, we explore the process of selecting the most suitable open-source embedding model for your Retrieval-Augmented Generation (RAG) application. \n",
+    "\n",
+    "We evaluate three open-source embedding models: [nomic-embed-text](https://ollama.com/library/nomic-embed-text), [bge-m3](https://ollama.com/library/bge-m3/blobs/daec91ffb5dd), and [mxbai-embed-large](https://ollama.com/library/mxbai-embed-large/blobs/819c2adf5ce6).\n",
+    " \n",
+    "To facilitate the execution of these models and the generation of embeddings, we will use the following tools:\n",
+    "\n",
+    "- [Ollama](https://ollama.com/), a platform that provides access to a variety of open-source Large Language Models (LLMs).\n",
+    "- [pgai](https://github.com/timescale/pgai), an open-source extension that seamlessly integrates LLM workflows, such as embedding creation and management, directly into your PostgreSQL database.\n",
+    "\n",
+    "The evaluation process involves:\n",
+    "\n",
+    "1. Setting up a test environment with Ollama and PostgreSQL\n",
+    "2. Loading Paul Graham's essays as our test dataset\n",
+    "3. Generating embeddings using different models\n",
+    "4. Creating diverse test questions across multiple categories\n",
+    "5. Evaluating each model's retrieval performance"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Environment Setup\n",
+    "\n",
+    "Before you begin, make sure to update the `OPENAI_API_KEY` in the `compose.yaml` file. Then execute the `install.sh` shell script to set up your environment. This script will:\n",
```

Comment:
I think it makes sense to mention why this is needed. I was a bit confused (open source embedding models? why do I need an openai_api_key?

Something like:
> We will use 4o as a "LLM as a judge" to evaluate our embedding results. To set that up we need and OPENAI_API_KEY.

Reply:
I removed it since we decided to use Llama 3.2, another open-source LLM, to keep everything on theme.